### ISSUE TITLE
Enhancement: [strict-boolean-expressions] allowNullableTrue

### ISSUE THREAD

**Issue by everhardt** [typescript-eslint/typescript-eslint#7759]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/strict-boolean-expressions

### Description

Since https://github.com/typescript-eslint/typescript-eslint/issues/4103 there is no warning raised when a type `true | undefined` is used  in expressions where a boolean is expected. I think that is inconsistent (`true | undefined` is not a boolean and we also allow disabling other "safe" options like `allowNullableObject`), but I accept that others like it that way. I propose to introduce `allowNullableTrue` (default `true`) that I can set to `false`, eg.:

```json
{
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": [
          "error",
          {
            "allowString": false,
            "allowNumber": false,
            "allowNullableObject": false,
            "allowNullableTrue": false,
          },
        ]
  }
}
```

### Fail

```typescript
function myFunction(a: true | undefined): void {
  if (a) {
    console.log('something')
  }
}
```

### Pass

```typescript
function myFunction(a: true | undefined): void {
  if (a === true) {
    console.log('something')
  }
}
```

**Comment by JoshuaKGoldberg** [comment]
👋 it's been over half a year since this was posted and we haven't seen any additional evidence of community desire for it. That leads us to believe that this is a pretty niche use case which we can't justify spending time creating/reviewing & then maintaining. `strict-boolean-expressions` already has a lot of options, including a somewhat-similar `allowNullableBoolean`. 

I'm going to go ahead and close this out as a wontfix. But if anybody feels strongly it should be reconsidered, please do post back in a new issue with a link to this one and an explanation of your use case. Cheers!
